tanner/api/api.py

```diff
@@ -34,25 +56,107 @@ async def return_snare_stats(self, snare_uuid):
         result['total_sessions'] = len(sessions)
         for sess in sessions:
             result['total_duration'] += sess['end_time'] - sess['start_time']
-            for attack in sess['attack_types']:
-                if attack in result['attack_frequency']:
-                    result['attack_frequency'][attack] += 1
+            for path in sess['paths']:
+                if path['attack_type'] in result['attack_frequency']:
+                    result['attack_frequency'][path['attack_type']] += 1
 
         return result
 
-    async def return_snare_info(self, uuid, count=-1):
+    async def return_snare_info(self, uuid):
+        """Returns JSON data that contains information about
+         all the sessions a single snare instance have.
+
+        Returns:
+            uuid [string] - Snare UUID
+        """
         query_res = []
-        try:
-            query_res = await self.redis_client.zrevrangebyscore(
-                uuid, offset=0, count=count, encoding='utf-8'
-            )
-        except aioredis.ProtocolError as connection_error:
-            self.logger.exception('Can not connect to redis %s', connection_error)
-        else:
-            if not query_res:
-                return 'Invalid SNARE UUID'
-            for (i, val) in enumerate(query_res):
-                query_res[i] = json.loads(val)
+        async with self.pg_client.acquire() as conn:
+            async with conn.cursor() as cur:
+                await cur.execute(
+                    """
+                    SELECT * FROM sessions
+                    WHERE
+                        sessions.sensor_id = '%s'
+                    """ % (uuid)
+                )
+                session = await cur.fetchall()
```

Comment:
I would suggest you to think about possibility to control the number of sessions you fetched, if you will have too many sessions inside one snare, fetching all might cause the problem.